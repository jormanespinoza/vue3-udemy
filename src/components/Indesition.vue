<template>
  <img v-if="image" :src="image" alt="bg" />
  <div class="bg-dark"></div>
  <div class="indesition-container">
    <input type="text" placeholder="Ask a question" v-model="question" />
    <p>Remember to close it with a question mark (?).</p>

    <div v-if="isValidQuestion">
      <h2>{{ question }}</h2>
      <h1>{{ answer }}</h1>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Indesition',
  data() {
    return {
      question: '',
      answer: null,
      image: null,
      isValidQuestion: false
    }
  },
  methods: {
    async getAnswer() {
      this.answer = 'Thinking...'
      const {answer, image} = await fetch('https://yesno.wtf/api').then(response => response.json())
      this.answer = answer
      this.image = image
      this.isValidQuestion = true
    }
  },
  watch: {
    question(value, oldValue) {
      this.isValidQuestion = false
      if (!value.includes('?')) return
      this.getAnswer()
    }
  }
}
</script>

<style scoped>
img,
.bg-dark {
  height: 100vh;
  left: 0px;
  max-height: 100%;
  max-width: 100%;
  position: fixed;
  top: 0px;
  width: 100vw;
}

.bg-dark {
  background-color: rgba(0, 0, 0, 0.4);
}

.indesition-container {
  position: relative;
  z-index: 1;
}

input {
  width: 250px;
  padding: 10px 15px;
  border-radius: 5px;
  border: none;
}
input:focus {
  outline: none;
}

p {
  color: white;
  font-size: 20px;
  margin-top: 0px;
}

h1,
h2 {
  color: white;
}

h2 {
  margin-top: 150px;
}
</style>