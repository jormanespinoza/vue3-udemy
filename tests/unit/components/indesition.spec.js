import { shallowMount } from '@vue/test-utils'
import Indesition from '@/components/Indesition'

describe('Counter.vue', () => {
  let wrapper
  let clgSpy
  global.fetch = jest.fn()

  beforeEach(() => {
    wrapper = shallowMount(Indesition)
    clgSpy = jest.spyOn(console, 'log')
    jest.clearAllMocks()
  })

  test('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should not do anything on input keypress', async () => {
    const getAnswerSpy = jest.spyOn(wrapper.vm, 'getAnswer')
    const input = wrapper.find('input')
    await input.setValue('Hello World')

    expect(clgSpy).toHaveBeenCalledTimes(1)
    expect(getAnswerSpy).not.toHaveBeenCalled()
  })

  test('should getAnswer on question mark (?) keypress', async () => {
    const getAnswerSpy = jest.spyOn(wrapper.vm, 'getAnswer')
    const input = wrapper.find('input')
    await input.setValue('Will I be good at Vue?')

    expect(clgSpy).toHaveBeenCalledTimes(1)
    expect(getAnswerSpy).toHaveBeenCalled()
  })


  test('should getAnswer', () => {

  })

  test('should fail on getAnswer', () => {

  })
})