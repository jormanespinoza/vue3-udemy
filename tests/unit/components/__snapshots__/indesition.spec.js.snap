// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue should match snapshot 1`] = `
<!--v-if-->
<div class="bg-dark"></div>
<div class="indesition-container"><input type="text" placeholder="Ask a question">
  <p>Remember to close it with a question mark (?).</p>
  <!--v-if-->
</div>
`;
